internal/scorer/algorithm/input_test.go

```diff
@@ -0,0 +1,184 @@
+package algorithm
+
+import "testing"
+
+func TestBoundsApply(t *testing.T) {
+	tests := []struct {
+		name string
+		b    Bounds
+		v    float64
+		want float64
+	}{
+		{
+			"regular test",
+			Bounds{
+				0,
+				10,
+				false,
+			},
+			7,
+			7,
+		},
+		{
+			"SmallerIsBetter equals true",
+			Bounds{
+				0,
+				10,
+				true,
+			},
+			7,
+			3,
+		},
+		{
+			"lower is not 0",
+			Bounds{
+				40,
+				80,
+				false,
+			},
+			50,
+			10,
+		},
+		{
+			"upper bound > lower bound", // should this test work?
+			Bounds{
+				40,
+				20,
+				false,
+			},
+			30,
+			0,
+		},
+		{
+			"upper bound == lower bound", // similar question as above, should this work?
+			Bounds{
+				40,
+				40,
+				false,
+			},
+			40,
+			0,
+		},
+		{
+			"upper bound == lower bound and SmallerIsBetter is true", // same question as above
+			Bounds{
+				40,
+				40,
+				true,
+			},
+			40,
+			0,
+		},
+		{
+			"v is negative", // can v be negative?
+			Bounds{
+				0,
+				10,
+				false,
+			},
+			-10,
+			0,
+		},
+		{
+			"v is lower than lower bound and SmallerIsBetter is true",
+			Bounds{
+				20,
+				30,
+				true,
+			},
+			15,
+			10,
+		},
+		{
+			"v is greater than upper bound and SmallerIsBetter is true",
+			Bounds{
+				0,
+				10,
+				true,
+			},
+			20,
+			0,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := test.b.Apply(test.v); got != test.want {
+				t.Errorf("Apply(%v) = %v, want %v", test.v, got, test.want)
+			}
+		})
+	}
+}
+
+func TestInputValue(t *testing.T) {
+	type fields struct {
+		Source       Value
+		Bounds       *Bounds
+		Distribution *Distribution
+		Tags         []string
+		Weight       float64
+	}
+	//nolint:govet
+	tests := []struct {
+		name   string
+		input  fields
+		fields map[string]float64
+		want   float64
+		canDo  bool
```

Comment:
Since these are both "want" fields I'd prefer either:

```go
wantValue float64
wantOk bool
```
or

```go
want struct{ value float64, ok bool }
```

Reply:
I have taken care of it. Thanks!